### DISCUSSION TITLE
Best practice for loading profiles and avatars?

### DISCUSSION THREAD

**Discussion by Jordan-Ellis** [supabase/supabase#18877] [Questions]
I’ve implemented user profiles as found in Supabase's docs. My question is what is the recommended way of retrieving that information in a performant way, and whether or not it’s a good idea to cache it. 

Every page reload, the user’s name and avatar image id is loaded.

```js
const { data, error } = await supabase.from('profiles').select(`full_name, avatar_url`).eq('id', user.value.id).single()
```

Then, the user’s avatar image url is loaded:

```js
const { data } = await supabase.storage.from('avatars').createSignedUrl(profile.value.avatarPath, 6000, {
    transform: {
        width: 100,
        height: 100,
    },
})
```

Since this requires two extra steps to get the users information every page load, is there a recommended way to cache this (such as cookies or local storage)? Or is that perfectly fine to do without being bad for performance? I'm also not sure if caching info like that is a bad security practice. 

It seems to me that if a user is using my app for multiple days in a row, then that information would be unnecessarily fetched many times.

Any suggestions would be appreciated!

**Comment by GaryAustin1** [comment]
You could also store the user's avatar path in the user_metadata so then you have it when you sign in.  Same with full_name.  
You could also have avatars as UUID's of the user.
Because you have them in a non-public bucket though you do have to call to get the signedUrl when you need it.

There are probably ways to cache the signedUrl if it is set to expire and I would think the browser would cache it.  You would have to deal with errors though if not cached.

You could store the avatar in localstorage and there is lots of info on the web for doing that.

**Reply by Jordan-Ellis** [reply]
Thanks! It seems that user_metadata is the best way of doing it I've seen so far.
